uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/leave-request-popup/leave-request-popup-body.html

```diff
@@ -29,7 +29,7 @@
           <select
             class="no-select2"
             ng-hide="$ctrl.absenceTypes.length === 0"
-            ng-disabled="!$ctrl.isRole('admin') && ($ctrl.isMode('view') || ($ctrl.isRole('manager') && !$ctrl.isMode('create')))"
+            ng-disabled="!$ctrl.isRole('admin') && ($ctrl.isMode('view') || ($ctrl.isRole('manager') && !$ctrl.isMode('create'))) || $ctrl.loading.entitlements"
```

Comment:
this condition is getting increasingly complicated, move it in the controller

Reply:
Not quite sure this is necessary, this is just a condition which is just a bit longer than usual.

Reply:
it's hard to follow. put it in a method, document it, split the individual conditions and put them in variables with meaningful names

Reply:
Done. I am not sure we need variables. I attempted to make it as readable as possible.